### ISSUE TITLE
SAMPLES assign projectID's

### ISSUE THREAD

**Issue by kaijagahm** [MFEh2o/db#94]
projectID:
Missing from all except SED_TRAP_SAMPLES, where we made a deliberate decision to add it.
Was this intentionally left out? I can see how maybe a given sample could be used for more than one project. But then again, that might also be true of other types of measurements. So I don't know that that by itself is a reason not to have a projectID column. Even in our discussion

lakeID:
Missing from all four tables.
I feel like we've discussed this before--it's not strictly necessary, since you can easily pull lakeID out of the sampleID. And it's a bit un-normalized to have it all over the place. It does make things easier, and in particular, the projectlakes/lakeprojects functions I wrote are pretty much useless without a lakeID column.

**Comment by kaijagahm** [comment]
If we add lakeID, then should also add it to WATER_CHEM.

**Comment by kaijagahm** [comment]
Added dateSample and lakeID to WATER_CHEM. Added lakeID to the other *SAMPLES tables.

Have not added projectID yet--going to do some investigation after fixing all the sampleID's.

**Comment by kaijagahm** [comment]
Waiting to proceed with this one until after #51 and #60 are done.

**Comment by kaijagahm** [comment]
SED_TRAP_SAMPLES already has projectID's. Was able to assign almost all for FISH_SAMPLES. Had a lot of trouble assigning projectID's for SAMPLES, because many of the methods are generic and/or used across multiple projects.

**Comment by kaijagahm** [comment]
Most updates made in 4.5.2. Still waiting on filling out the projectID column in SAMPLES--it currently has a lot of NA's because I had trouble clearly assigning projectID's based on metadataID's.

**Comment by kaijagahm** [comment]
Made this google sheet: https://docs.google.com/spreadsheets/u/1/d/11lJ4nCTZ59nvwivLbyocO6Bu0ENaVYNzLDpHKwLNR-s/edit?urp=gmail_link&gxids=7628#gid=0

Agreed at 4/13 meeting that @joneslabND is going to take a look and propose solutions for as many of these as possible. Will be useful to have projectID's included in SAMPLES when we work on the entry tools.

**Comment by Randinotte** [comment]
I made a pass through the google sheet above and incorporated Stuart's slack comment that CR, HB, and MO are projectID 15, other lakes are projectID 17 (for regular limno-type samples), and FE/WL are projectID 3. 

Re: "GC weirdness" rows, when samples from multiple projectID's are run in the same batch on the GC (which is very common), the standards used to calibrate the GC apply to both projectIDs in the batch. So realistically, it's arbitrary which projectID gets assigned to the HankLab samples. The only conflict with picking one randomly over the other is that if someone wanted to "pull all the GC data from projectID 17" they might not get all the standards pulled with it (if they were randomly assigned to another projectID because they were in a batch with something else).

**Comment by kaijagahm** [comment]
@joneslabND any chance you'll have time to get to this in the next couple of days? It would be convenient to incorporate it into the database update I'm preparing to do. If you can't, that's fine--just let me know if I should wait for you.

**Comment by joneslabND** [comment]
Sorry, I was in the field at UNDERC all week. Have you moved ahead with the update? I could get to this by the middle of next week, but it is ok to do an update without this if you'd like to (or already have) finish the update.

**Comment by kaijagahm** [comment]
That's okay! I've already done the update, so it can wait for the next one.

**Comment by kaijagahm** [comment]
Stuart resolved all the conflicting projectID's:
[conflictingProjectIDs.20210406_SEJ_edit.csv](https://github.com/MFEh2o/db/files/6735549/conflictingProjectIDs.20210406_SEJ_edit.csv)

Incorporated this into v 4.7.0 of the database and assigned the rest of the projectID's. Just missing a few more with routine limno sampling metadataID's.
